zebra/zebra_evpn_mh.c

```diff
@@ -1694,8 +1694,9 @@ void zebra_evpn_es_cleanup(void)
 
 	RB_FOREACH_SAFE(es, zebra_es_rb_head,
 			&zmh_info->es_rb_tree, es_next) {
-		zebra_evpn_local_es_del(es);
```

Comment:
Code looks fine I'll just suggest two nits:

1. It seems to be a trend to use "free" functions which take a pointer to a pointer so you can `free()` the memory and already set the pointer to `NULL`. Your case fits even better because then you don't need to do a return value.

   Example:

   ```c
   static void zebra_evpn_es_free(struct zebra_evpn_es **esp)
   {
   	struct zebra_evpn_es *es = *esp;

   	/* Convinience NULL check: */
   	if (es == NULL)
   		return;

   	if ((es->flags & (ZEBRA_EVPNES_LOCAL | ZEBRA_EVPNES_REMOTE)) ||
   			listcount(es->mac_list))
   		return;

   	/* ...code here... */

   	/* XFREE will already set (*esp) to `NULL`. */
   	XFREE(MTYPE_ZES, (*esp));
   }
   ```

   Benefits:
   a. Never use a pointer after free again
   b. Double `free()`s will be harder to get (unless we leave a pointer dangling somewhere else)
   c. Less checks outside these functions for your use case

   Drawbacks:
   a. There will be a bit more of code changes to accomplish this here.

2. Also use `zebra_evpn_remote_es_flush` with same technique and now you no longer need the `if (es)` below.

Reply:
Yes, This is definitely more elegant. I will fixup.

PS: During some aggressive cleanup I accidentally cleaned up the branch in my fork (still have it locally, phew!). I will open a new PR with this fixup included.